scripts/generate_conda_file.py

```diff
@@ -74,6 +75,7 @@
     "hyperopt": "hyperopt==0.1.1",
     "idna": "idna==2.7",
     "memory-profiler": "memory-profiler>=0.54.0",
+    "nni": "nni==0.5.2.1"
```

Comment:
so, as I mentioned somewhere, do we want to include this in the env? Our env is getting bigger and thus quite slow to install & activate it now.

One option is to remove not-very-common modules (e.g. `hyperopt` `nni` `memory-profiler`) and add a prerequisite step to each notebook that has dependency on such packages. Any thoughts? @miguelgfierro @yueguoguo

Reply:
BTW, it seems there is a comma missing here.

Reply:
I agree with @loomlike and it indeeds affects user experience in some way. It would be great to have a finer granularity of yaml files for different workloads. E.g., we may add options to yaml generation script to enable a desired env.

Reply:
I think there is going to be a trade between how big the dependencies are vs how difficult to generate the conda file if we start adding many options. Many dependencies will make that when we install the env it takes longer, for the tests this can be annoying. On the other hand, it is easier for the user to have a simple conda generator so they can just do two steps: generate conda file and install conda file.

You know what I'll do? I will first make sure that we are using all the libraries in the generator, for example, are we using dask? Also, we are using gitpython and pymongo for tracking the metrics, but this is not related to the utilities. Maybe it would make sense to have an env file in the tracking metrics folder with the libraries that we need there. 
What do you guys think

Reply:
Definitely, for the tracking metrics the conda env could be different. I like Le's suggestion of using flags for the `generate_conda.py` script e.g. `--nni` would allow one to use NNI. That said, if we remove tracking metrics and debugging from the conda env, we are not left with many optional libraries other than NNI and hyperopt. For example, there is no rationale for separating recommendation algorithms, so you cannot e.g. make fastai an option.

Reply:
> I like Le's suggestion of using flags for the generate_conda.py script e.g. --nni would allow one to use NNI

The danger we have with this is to end up with a super complex conda generator. Now we have 3 options: base, --gpu and --pyspark. If we start adding options for libraries I think for the user it will be more complicated. Imagine that we have the --nni option, I'm sure users will start installing the repo, will forget about the --nni option and will have to go back to the readme to see what happens

Reply:
Good point, it doesn't affect many libraries anyway. Also, once you have created the conda environment, it is fast to update it.